pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,443 @@
+	})
+}
+
+func setVerbosity(currentLv, options map[string]uint, patchData *[]patch.PatchOperation) {
+	// update currentLv based on the user-specified verbosity for all components
+	if verbosity, exist := options["all"]; exist {
+		for component := VirtAPI; component < All; component++ {
+			JSONName := getJSONNameByVirtComponent(component)
+			currentLv[JSONName] = verbosity
+		}
+	}
+
+	// update currentLv based on the user-specified verbosity for each component
+	for componentName, verbosity := range options {
+		if componentName == "all" {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		currentLv[JSONName] = verbosity
+	}
+
+	// in case of just reset (no set operation after the reset), don't need to add another patch
+	if len(currentLv) != 0 {
+		addPatch(patchData, currentLv)
+	}
+}
+
+func createPatch(currentLv, options map[string]uint) ([]byte, error) {
+	patchData := []patch.PatchOperation{}
+
+	// reset only if verbosity exists, otherwise do nothing
+	if isReset && len(currentLv) != 0 {
+		// add an empty object
+		currentLv = map[string]uint{}
+		addPatch(&patchData, currentLv)
+	}
+
+	setVerbosity(currentLv, options, &patchData)
+
+	return json.Marshal(patchData)
+}
+
+func findOperation(cmd *cobra.Command, options map[string]uint) (operation, error) {
+	isShow, isSet := false, false
+
+	for component := VirtAPI; component <= All; component++ {
+		componentName := GetComponentNameByVirtComponent(component)
+
+		// check if the flag for the component is specified
+		if !cmd.Flags().Changed(componentName) {
+			continue // do nothing for the component
+		}
+
+		// if flag is specified, it means either set or show
+		// if the value = noArg, it means show
+		// if the value != noArg, it means set
+		isShow = isShow || verbosities[component] == noArg
+		isSet = isSet || verbosities[component] != noArg
+
+		// check whether the verbosity is in the range
+		// allow noArg (=10)
+		// because user can see the value 10 as the default in the help like this:
+		// 		--all uint[=10] show/set all component log verbosity (0-9) (default 11)
+		// --all=10, which equals -all (show operation)
+		if verbosities[component] != noArg && verbosities[component] > maxVerbosity {
+			return nop, fmt.Errorf("%s: log verbosity must be %d-%d", componentName, minVerbosity, maxVerbosity)
+		}
+
+		// add a componentName and its verbosity to the map
+		// in case of show: verbosity = noArg
+		// in case of set: verbosity = 0-9
+		// in case of reset: no information added to the map
+		options[componentName] = verbosities[component]
+	}
+
+	// do not distinguish between set and reset at this point, because set and reset can coexist
+	if isReset {
+		isSet = true
+	}
+
+	if isShow && isSet {
+		return nop, errors.New("only show or set is allowed")
+	}
+
+	if isShow {
+		return show, nil
+	} else if isSet {
+		return set, nil
+	}
+
+	return nop, nil
+}
+
+func (c *Command) RunE(cmd *cobra.Command) error {
+	// get client
```

Comment:
nit: superfluous comment, it is clear from the code

Reply:
Agreed. Will remove it.